src/datachain/lib/file.py

```diff
@@ -417,14 +418,30 @@ def save(self, destination: str):
         self.read().save(destination)
 
 
-class IndexedFile(DataModel):
-    """Metadata indexed from tabular files.
-
-    Includes `file` and `index` signals.
-    """
+class ArrowVFile(DataModel):
+    """`DataModel` for reading arrow files."""
 
     file: File
     index: int
+    kwargs: dict
+
+    @contextmanager
+    def open(self):
+        """Stream row contents from indexed file."""
+        if self.file._caching_enabled:
+            path = self.file.get_local_path(download=True)
+            ds = dataset(path, **self.kwargs)
+
+        else:
+            path = self.file.get_path()
+            ds = dataset(path, filesystem=self.file.get_fs(), **self.kwargs)
+
+        return ds.take([self.index]).to_reader()
+
+    def read(self):
+        """Returns row contents."""
+        with self.open() as record_batch:
+            return record_batch.to_pylist()[0]
```

Comment:
should it be index here instead of `[0]`?

Reply:
No, we already take the index above in the open method. This is always a list of one and this ensures we return the dictionary instead of a list containing the dictionary.